snuba/datasets/dataset.py

```diff
@@ -44,17 +46,22 @@ class Dataset(object):
 
     # TODO: Still not sure if the Dataset class really needs the actual Entity object
     # or just the name.
-    def __init__(self, *, default_entity: Entity) -> None:
+    def __init__(self, *, default_entity: EntityKey) -> None:
         # TODO: This is a convenience while we slowly migrate everything to Entities. This way
         # every dataset can have a default entity which acts as a passthrough until we can
         # migrate the datasets to proper entities.
-        self.__default_entity = default_entity
+        self.default_entity_key = default_entity
+        self.__default_entity = get_entity(default_entity)
 
     def get_entity(self, entity_name: Optional[str]) -> Entity:
         return self.__default_entity
 
+    # TODO: Remove once entity selection moves to Sentry
+    def select_entity(self, query: Query) -> EntityKey:
+        return self.default_entity_key
+
     def get_query_plan_builder(
-        self, entity_name: Optional[str] = ""
+        self, entity_name: Optional[str] = None
```

Comment:
This method should not be needed in the dataset anymore.

Reply:
Updated the remaining tests and removed it

Reply:
To clarify, why is this not needed? It will definitely be needed for JOINs. Or is the plan to add it back once the work on JOINs starts?

Reply:
It's not needed now because the entity that is initially picked is stored in the query and used from there to get the query plan and the rest of the query pipeline. I'm not sure how joins will work exactly, though I'd assume the join structure selected either needs to be stored in the query also or passed around through the query processing stages so all parts of the pipeline have access to it - and the query plan would be provided by that join object, not the dataset. I guess you might have had something slightly different in mind though that goes through the dataset instead?

Reply:
@evanh you are right, it will be needed with joins, though the return type and the role will be fairly different, we can add it back then. Now that this PR is adding the referenced entity to the query, the query processing pipeline can rely on that information to pick the plan and it should not rely on dataset anymore for a plan (until we add joins), so leaving it here when it just returns the one for an entity seems confusing (the entity already provides the same thing).